datasource.go

```diff
@@ -31,19 +31,10 @@ type DataSource struct {
 }
 
 // JSONData is a representation of the datasource `jsonData` property
-type JSONData struct {
-	AssumeRoleArn           string `json:"assumeRoleArn,omitempty"`
-	AuthType                string `json:"authType,omitempty"`
-	CustomMetricsNamespaces string `json:"customMetricsNamespaces,omitempty"`
-	DefaultRegion           string `json:"defaultRegion,omitempty"`
-	TlsSkipVerify           bool   `json:"tlsSkipVerify,omitempty"`
-}
+type JSONData map[string]interface{}
```

Comment:
I'm curious to hear your reasoning for changing the types here. Do we have to worry about any potential consumers of Datasource method since they'll have to change how they access the JSONData and SecureJSONData properties?

Reply:
There are multiple DataSource types available for Grafana and struct here allows only to use go-grafana-api only with one DataSource (I assume it was something with AWS).

If we loosen typing here to `map[string]interface{}` we are able to pass different jsonData and secureJsonData needed for other datasource types.

We know breaking changes are not welcomed into the interfaces and we are more than happy to make this possible without breaking changes but I don't see a way how to achieve that. Maybe with reflect or changing type to some Inteface but I am not a fan of either.

Reply:
Breaking changes aside, I'm not sure it's necessary to remove the schema from `jsonData` and `secureJsonData` entirely to support additional data source types. There tends to be some overlap in values between the different data sources; how we've handled different schemas for similar objects is to merge the maps of all the possible keys of the different data source types and tack on the `omitempty` tag to most of them so that they won't be sent unless explicitly used. Does that sound like an alright approach to you?